piplmesh/account/backends.py

```
@@ -45,41 +45,28 @@ def authenticate(self, facebook_token=None, request=None):
-    
-        # Retrieve access token
-        url = urllib.urlopen('https://graph.facebook.com/oauth/access_token?%s' % urllib.urlencode(args)).read()
-        response = urlparse.parse_qs(url)
-        access_token = response['access_token'][-1]
-    
-        # Retrieve user's public profile information
-        data = urllib.urlopen('https://graph.facebook.com/me?access_token=%s' % access_token)
-        fb = json.load(data)
 
+        fb, access_token = getFBData(facebook_token, request, 'facebook_callback')
         # TODO: Check if id and other fields are returned
         # TODO: Move user retrieval/creation to User document/manager
         # TODO: get_or_create implementation has in fact a race condition, is this a problem?
-        user, created = self.user_class.objects.get_or_create(
-            facebook_id=fb.get('id'),
-            defaults={
-                'username': fb.get('username', fb.get('first_name') + fb.get('last_name')),
-                'first_name': fb.get('first_name'),
-                'last_name': fb.get('last_name'),
-                'email': fb.get('email'),
-                'gender': fb.get('gender'),
-                'facebook_link': fb.get('link'),
-            }
-        )
-        user.facebook_token = access_token
-        user.save()
-
-        return user
+        try:
+            user, created = self.user_class.objects.get_or_create(
+                facebook_id=fb.get('id'),
+                defaults={
+                    'username': fb.get('username', fb.get('first_name') + fb.get('last_name')),
+                    'first_name': fb.get('first_name'),
+                    'last_name': fb.get('last_name'),
+                    'email': fb.get('email'),
+                    'gender': fb.get('gender'),
+                    'facebook_link': fb.get('link'),
+                    }
+            )
+            user.facebook_token = access_token
+            user.save()
+            return user
+        except Exception:
```

Never ever just hide exceptions. No. You do not want this.